plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -343,10 +356,23 @@ public function addDocument($data, $type)
         unset($data['id']);
 
         $document = new \Elastica\Document($id, $data);
-        $document->setType($type);
+
+        // Setting a dummy type since it is required in ES 6.x
+        // but it can be removed in 7.x when it becomes optional
+        $document->setType(self::ES_TYPE);
+
+        if (!$this->currentBatchType) {
+            $this->currentBatchType = $type;
```

Comment:
We may even want to change the param name to `$indexName` instead of `$type`.

Reply:
Also, unrelated to these changes, shouldn't this function be checking if it's enabled? Like all the others updates and delete.